example/idl/thrift does not compile in YARP devel

~~~
traversaro@traversaro-VirtualBox:~/src/codyco-superbuild/libraries/YARP/example/idl/thrift/build$ CMAKE_PREFIX_PATH=/home/traversaro/src/codyco-superbuild/build/install cmake .
Thrift headers include/firstInterface/PointQuality.h;include/firstInterface/PointD.h;include/firstInterface/PointDLists.h;include/firstInterface/PointWithQuality.h;include/secondInterface/Demo.h;include/secondInterface/DemoExtended.h;include/secondInterface/demo_common.h.
 Thirft sources: src/firstInterface/PointQuality.cpp;src/firstInterface/PointD.cpp;src/firstInterface/PointDLists.cpp;src/firstInterface/PointWithQuality.cpp;src/secondInterface/Demo.cpp;src/secondInterface/DemoExtended.cpp
-- Configuring done
-- Generating done
-- Build files have been written to: /home/traversaro/src/codyco-superbuild/libraries/YARP/example/idl/thrift/build
traversaro@traversaro-VirtualBox:~/src/codyco-superbuild/libraries/YARP/example/idl/thrift/build$ make
Scanning dependencies of target generatedDemoLibrary
[  6%] Building CXX object generated/CMakeFiles/generatedDemoLibrary.dir/src/firstInterface/PointQuality.cpp.o
/home/traversaro/src/codyco-superbuild/libraries/YARP/example/idl/thrift/generated/src/firstInterface/PointQuality.cpp:4:26: fatal error: yarp/os/Wire.h: No such file or directory
compilation terminated.
generated/CMakeFiles/generatedDemoLibrary.dir/build.make:62: recipe for target 'generated/CMakeFiles/generatedDemoLibrary.dir/src/firstInterface/PointQuality.cpp.o' failed
make[2]: *** [generated/CMakeFiles/generatedDemoLibrary.dir/src/firstInterface/PointQuality.cpp.o] Error 1
CMakeFiles/Makefile2:103: recipe for target 'generated/CMakeFiles/generatedDemoLibrary.dir/all' failed
make[1]: *** [generated/CMakeFiles/generatedDemoLibrary.dir/all] Error 2
Makefile:83: recipe for target 'all' failed
make: *** [all] Error 2
~~~

--- a/example/idl/thrift/generated/CMakeLists.txt
+++ b/example/idl/thrift/generated/CMakeLists.txt
@@ -18,4 +18,5 @@ set(generated_libs_include_dir ${CMAKE_CURRENT_SOURCE_DIR}/include CACHE INTERNA
 include_directories(${generated_libs_include_dir})
 
 add_library(generatedDemoLibrary ${THRIFT_HEADERS} ${THRIFT_SOURCES})
+target_link_libraries(generatedDemoLibrary ${YARP_LIBRARIES})
 set_target_properties(generatedDemoLibrary PROPERTIES LINKER_LANGUAGE CXX)

--- a/example/matrix/CMakeLists.txt
+++ b/example/matrix/CMakeLists.txt
@@ -10,11 +10,13 @@ project(Matrix)
 
 find_package(YARP REQUIRED)
 
+find_package(GSL REQUIRED)
+
 include_directories(${CMAKE_SOURCE_DIR})
 
 add_executable(basic_ops basic_ops.cpp)
 target_link_libraries(basic_ops ${YARP_LIBRARIES})
 
 add_executable(gsl_example gsl_example.cpp)
-target_link_libraries(gsl_example ${YARP_LIBRARIES})
+target_link_libraries(gsl_example ${YARP_LIBRARIES} YARP::YARP_gsl GSL::gsl GSL::gslcblas)
 

--- a/example/matrix/gsl_example.cpp
+++ b/example/matrix/gsl_example.cpp
@@ -18,8 +18,10 @@
 
 #include <yarp/sig/Matrix.h>
 #include <yarp/sig/Vector.h>
+#include <yarp/gsl/Gsl.h>
 
 using namespace yarp::sig;
+using namespace yarp::gsl;
 
 int main(int argc, const char **argv)
 {
@@ -29,8 +31,8 @@ int main(int argc, const char **argv)
     v2=1;
 
     double r;
-    gsl_blas_ddot((gsl_vector *)(v1.getGslVector()),
-                  (gsl_vector *)(v2.getGslVector()), &r);
+    gsl_blas_ddot((gsl_vector *)GslVector(v1).getGslVector(),
+                  (gsl_vector *)GslVector(v2).getGslVector(), &r);
 
     printf("Res vector: %lf\n", r);
 
@@ -51,10 +53,10 @@ int main(int argc, const char **argv)
 
     gsl_blas_dgemm(CblasNoTrans, CblasNoTrans,
         1.0,
-        (const gsl_matrix *) m1.getGslMatrix(),
-        (const gsl_matrix *) m2.getGslMatrix(),
+        (const gsl_matrix *) GslMatrix(m1).getGslMatrix(),
+        (const gsl_matrix *) GslMatrix(m2).getGslMatrix(),
         0.0,
-        (gsl_matrix *) m3.getGslMatrix());
+        (gsl_matrix *)GslMatrix(m3).getGslMatrix());
 
     printf("Result: (%s)", m3.toString().c_str());
 

